Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -696,6 +841,7 @@ namespace AzToolsFramework
                         // CreateWidgetForHandler will add a new entry to m_widgetToPropertyHandlerInfo, kill the old entry
                         GetDPE()->ReleaseHandler(AZStd::move(foundEntry->second.hanlderInterface));
                         m_widgetToPropertyHandlerInfo.erase(foundEntry);
+                        m_columnLayout->RemoveSharePriorColumn(childIndex);
```

Comment:
this handles the morph case, and I see that AddColumnWidget handles the add case, but what about if the attributes just change? Like, what if the handler type stays the same, but suddenly it wants to sharePriorColumn? Is that handled somewhere I'm missing? It seems like there's code missing below in the area of "// handler is the same, set the existing handler with the new value"

Reply:
Yeah I wasn't sure whether we needed to add attribute checks there or not, since I wasn't sure of instances where attributes can change if the handler type stays the same. I opted not to because it was using that code path a lot and we were making those checks every single time but never doing anything. 
The SetPropertyEditorAttributes method can handle checking attributes for a domValue and a domIndex, so it just be added there :)